tuf/tuf.go

```diff
@@ -173,6 +173,132 @@ func (tr *Repo) RemoveBaseKeys(role string, keyIDs ...string) error {
 	return nil
 }
 
+// GetBaseRole gets a base role from this repo's metadata
+func (tr *Repo) GetBaseRole(name string) (data.BaseRole, error) {
+	if !data.IsBaseRole(name) {
+		return data.BaseRole{}, fmt.Errorf("Invalid base role name specified: %s", name)
+	}
+	roleData, ok := tr.Root.Signed.Roles[name]
+	if !ok {
+		return data.BaseRole{}, fmt.Errorf("TUF data for %s does not exist in root metadata", name)
+	}
+	// Get all public keys for the base role from TUF metadata
+	keyIDs := roleData.KeyIDs
+	pubKeys := make(map[string]data.PublicKey)
+	for _, keyID := range keyIDs {
+		pubKey, ok := tr.Root.Signed.Keys[keyID]
+		if !ok {
+			return data.BaseRole{}, fmt.Errorf("key %s does not exist in root metadata", keyID)
```

Comment:
Nitpick: Could we add the role to the error message?  Maybe something like `"key with ID %s, specified as a signing key for role %s, was not found in root metadata"`?